### PR TITLE
Add golden test for `cardano-testnet` help

### PR DESCRIPTION
**Pull request by newhoggy**
# Description

We care about the help usage of the `cardano-testnet` as well, for example: https://github.com/input-output-hk/cardano-node/pull/5234, so it's good to have golden tests for them.

Depends on https://github.com/input-output-hk/cardano-node/pull/5206

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

**Review by erikd — APPROVED** [review]

File: cardano-cli/cardano-cli.cabal [hunk 1]
```diff
@@ -174,28 +174,41 @@ executable cardano-cli
...
 
   build-depends:        aeson
                       , base16-bytestring
+                      , bech32 >= 1.1.0
                       , bytestring
-                      , bech32            >= 1.1.0
                       , cardano-api ^>= 8.1.0.1
                       , cardano-api-gen ^>= 8.1.0.2
                       , cardano-cli
+                      , cardano-cli:test-lib
```
Comment (carbolymer) [inline cardano-cli/cardano-cli.cabal:204]: I think we can build `cabal-version` when we're using newer cabal features.

Comment (newhoggy) [reply cardano-cli/cardano-cli.cabal:204]: This feature was introduced in `cabal 3.0` and many other projects that this feature are still on `cabal-version: 3.0`.

Comment (carbolymer) [reply cardano-cli/cardano-cli.cabal:204]: https://cabal.readthedocs.io/en/stable/cabal-package.html#library
>For packages using cabal-version: 3.4 or higher, the syntax to specify an internal library in a build-depends: section is package-name:internal-library-name.

**Review by Jimbo4350 — APPROVED** [review]
Nice work :+1:

File: cardano-testnet/cardano-testnet-golden/data/golden/help/byron_genesis_COMMAND.cli [hunk 2]
```diff
@@ -1,5 +0,0 @@
-Invalid argument `COMMAND'
```
[the remark this answers is not in the document]
Comment (newhoggy) [reply cardano-testnet/cardano-testnet-golden/data/golden/help/byron_genesis_COMMAND.cli:1]: Code has been modified to not generate this file.

**Pull request merged**